src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -1823,6 +1824,11 @@ mongoc_collection_insert_one (
                                        reply,
                                        error,
                                        "insertedCount");
+   
+   // Only record _id of document if it was actually inserted.
+   if (result.nInserted > 0) {
```

Comment:
```suggestion
   if (reply && result.nInserted > 0) {
```

`reply` [may be NULL](https://mongoc.org/libmongoc/current/mongoc_collection_insert_one.html) if the caller does not want the `reply`. This may resolve some task failures in Evergreen.